File: backend/ms-users/src/main/java/pi/ms_users/domain/Payment.java
package pi.ms_users.domain;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonProperty.Access;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Entity
@Table(name = "Payment")
public class Payment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Enumerated(EnumType.STRING)
    @Column(name = "currency", nullable = false)
    private PaymentCurrency paymentCurrency;

    @Column(name = "amount", nullable = false)
    private BigDecimal amount;

    @Column(name = "date", nullable = false)
    private LocalDateTime date;

    @Column(name = "description", nullable = true)
    private String description;

    @Enumerated(EnumType.STRING)
    @Column(name = "concept", nullable = false)
    private PaymentConcept concept;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "contract_id", nullable = false, foreignKey = @ForeignKey(name = "fk_payment_contract"))
    private Contract contract;

    @ManyToOne(fetch = FetchType.LAZY, optional = true)
    @JoinColumn(name = "contract_utility_id", foreignKey = @ForeignKey(name = "fk_payment_contract_utility"))
    private ContractUtility contractUtility;

    @ManyToOne(fetch = FetchType.LAZY, optional = true)
    @JoinColumn(name = "commission_id", foreignKey = @ForeignKey(name = "fk_payment_commission"))
    private Commission commission;
}

File: frontend/src/app/user/types/payment.ts
export enum PaymentCurrency {
  USD = "USD",
  ARS = "ARS",
}

export enum PaymentConcept {
  ALQUILER = "ALQUILER",
  EXTRA = "EXTRA",
  COMISION = "COMISION",
}

export interface Payment {
  id: number;
  paymentCurrency: PaymentCurrency;
  amount: number;
  date: string; // ISO date-time
  description: string;
  concept: PaymentConcept;
  contractId: number;
  contractUtilityId?: number | null;
  commissionId?: number | null;
}

export interface PaymentCreate {
  paymentCurrency: PaymentCurrency;
  amount: number;
  date: string; // ISO date-time
  description: string;
  concept: PaymentConcept;
  contractId: number;
  contractUtilityId?: number;
  commissionId?: number;
}

export interface PaymentUpdate {
  id: number;
  paymentCurrency: PaymentCurrency;
  amount: number;
  date: string;
  description: string;
  concept: PaymentConcept;
  contractId: number;
  contractUtilityId?: number;
  commissionId?: number;
}
